leader-election-impl/src/main/java/com/palantir/leader/LeadershipEvents.java

```diff
@@ -88,6 +89,13 @@ void proposalFailure(PaxosRoundFailureException paxosException) {
         leaderElectionServiceMetrics.proposalFailure().mark();
     }
 
+    public void leaderOnOlderTimeLockVersion(OrderableSlsVersion version) {
+        // TODO(snanda): Kill log after few successful runs of blue-green deployment.
+        leaderLog.info("We contacted the leader and it reported that it is on an older version of TimeLock - {}",
+                withContextArgs(SafeArg.of("version", version)));
+        leaderElectionServiceMetrics.leaderOnOlderTimeLockVersion().mark();
```

Comment:
I'd suggest we rate limit the log message to maybe once every 10 or 30 seconds - timelocks otherwise ping each other 10 times per second meaning it'll be really noisy!